### PR TITLE
SSCS-10810 - Don't send relatedParties payload for Appointees, Jp's & Other parties

### PR DESCRIPTION
**Pull request by DGordonV1**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-10810
### Change description ###
Updates in hearing api to check for INTERPRETER role and process accordingly for HearingsDetailsMappings
**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by DGordonV1** [comment]
Quick comment, I had to include the excessive imports suppressed warning, I'd guess the way to resolve the PMD flag would be to use a wildcard for the "import static uk.gov.hmcts.reform.sscs.model.hmc.reference.EntityRoleCode" imports, otherwise we could refactor and move this method to a new class.

**Review by mcareyma — DISMISSED** [review]
This looks good. I can understand the point of the warning generally, but in this case we want which to be explicit about EntityRoleCodes we are importing.

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsMapping.java [hunk 1]
```diff
@@ -32,10 +33,12 @@
 import static uk.gov.hmcts.reform.sscs.helper.mapping.HearingsRequestMapping.buildHearingRequestDetails;
 import static uk.gov.hmcts.reform.sscs.model.hmc.reference.EntityRoleCode.APPELLANT;
 import static uk.gov.hmcts.reform.sscs.model.hmc.reference.EntityRoleCode.APPOINTEE;
+import static uk.gov.hmcts.reform.sscs.model.hmc.reference.EntityRoleCode.INTERPRETER;
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsMapping.java:36]: Could probably just * these

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java [hunk 2]
```diff
@@ -307,16 +307,10 @@ public static List<RelatedParty> getIndividualRelatedParties(Entity entity, Stri
         List<RelatedParty> relatedParties = new ArrayList<>();
         EntityRoleCode roleCode = getEntityRoleCode(entity);
         switch (roleCode) {
-            case APPOINTEE:
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java:310]: Shouldn't there be failing tests due to these deletions?

**Review by dattipoeHMCTS — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java (hunk 2 above)
Comment (DGordonV1) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java:310]: The only existing test was to check the REPRESENTATIVE returns an expected partyId and relationshipType (SOLICITOR). No coverage on the removed. I've added test for the new INTERPETER role.

**Review by dattipoeHMCTS — DISMISSED** [review]

**Review by TeaWithLucas — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsMapping.java (hunk 1 above)
Comment (ikirsanov) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsMapping.java:37]: dattipoeHMCTS, I think I had some of 'checkings' which told me to throw wildcards away

**Review by dattipoeHMCTS — APPROVED** [review]

**Review by ikirsanov — APPROVED** [review]

**Review by TeaWithLucas — APPROVED** [review]

**Pull request merged**